src/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -241,6 +241,47 @@ public static void ZipWithLargeSparseFile()
                 }
             }
         }
+
+        internal static readonly byte[] EmptyFileCompressed =
+           {
+            0x50, 0x4B, 0x03, 0x04, 0x14, 0x00, 0x06, 0x00, 0x08, 0x00, 0x00, 0x00, 0x21, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x16, 0x00, 0x00, 0x00, 0x78, 0x6C,
+            0x2F, 0x63, 0x75, 0x73, 0x74, 0x6F, 0x6D, 0x50, 0x72, 0x6F, 0x70, 0x65, 0x72, 0x74, 0x79, 0x32,
+            0x2E, 0x62, 0x69, 0x6E, 0x03, 0x00, 0x50, 0x4B, 0x01, 0x02, 0x14, 0x00, 0x14, 0x00, 0x06, 0x00,
+            0x08, 0x00, 0x00, 0x00, 0x21, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x16, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x78, 0x6C, 0x2F, 0x63, 0x75, 0x73, 0x74, 0x6F, 0x6D, 0x50, 0x72, 0x6F,
+            0x70, 0x65, 0x72, 0x74, 0x79, 0x32, 0x2E, 0x62, 0x69, 0x6E, 0x50, 0x4B, 0x05, 0x06, 0x00, 0x00,
+            0x00, 0x00, 0x01, 0x00, 0x01, 0x00, 0x44, 0x00, 0x00, 0x00, 0x36, 0x00, 0x00, 0x00, 0x00, 0x00
+            };
+
+        [Fact]
+        public static void ReadArchive_WithEmptyDeflatedFile()
+        {
+            using (var testStream = new MemoryStream(EmptyFileCompressed))
+            {
+                var tmpfilename = "xl/customProperty2.bin";
+                var tmppathname = Path.GetTempPath();
+                var tmpfilepath = tmppathname + tmpfilename;
+                // open archive with zero-length file that is compressed (Deflate = 0x8)
+                using (var zip = new ZipArchive(testStream, ZipArchiveMode.Update, true))
+                {
+                    // dispose without making any changes will rewrite the archive
+                }
+
+                byte[] fileContent = testStream.ToArray();
+
+                // compression method should stay the same (Deflate = 0x8)
+                Assert.Equal(8, fileContent[8]);
+
+                // extract and check the file. should stay empty.
+                using (var zip = new ZipArchive(testStream, ZipArchiveMode.Update))
+                {
+                    zip.ExtractToDirectory(tmppathname, overwriteFiles: true);
```

Comment:
Don't leak temp files in the tests.  There is no reason you need to test with ExtractToDirectory, just enumerate entries, assert that lengths are zero, then open the streams and ensure that lengths are zero.